tests/test_conflicts.py

```diff
@@ -44,94 +51,40 @@ def test_merge_conflicts(data, data_working_copy, geopackage, cli_runner, update
-        assert feature_name(sample_pks[2]) in r.stdout
-        assert feature_name(sample_pks[3]) in r.stdout
-        assert feature_name(sample_pks[4]) in r.stdout
+        r = cli_runner.invoke(["merge", "alternate", "--dry-run", f"--{output_format}"])
 
-        # Only modified in master:
-        assert feature_name(sample_pks[5]) not in r.stdout
-
-        r = cli_runner.invoke(["merge", "alternate"], input="a\no\nt\n")
-        assert r.exit_code == 0, r
-        assert repo.head.target.hex != alternate_commit_id
-        assert repo.head.target.hex != master_commit_id
-
-        r = cli_runner.invoke(["diff", master_commit_id])
         assert r.exit_code == 0, r
-
-        # Changed: their version is merged in from alternate automatically:
-        assert feature_name(sample_pks[0]) in r.stdout
-
-        # Not changed: our version is the same as their version:
-        assert feature_name(sample_pks[1]) not in r.stdout
-        # Changed: reverted to ancestor using "a":
-        assert feature_name(sample_pks[2]) in r.stdout
-        # Not changed: we kept our version using "o":
-        assert feature_name(sample_pks[3]) not in r.stdout
-        # Changed: accepted their version using "t":
-        assert feature_name(sample_pks[4]) in r.stdout
-        # Not changed: our version is merged in automatically:
-        assert feature_name(sample_pks[5]) not in r.stdout
-
-
-@pytest.mark.parametrize(
-    "data",
-    [
-        pytest.param(H.POINTS, id="points",),
-        pytest.param(H.POLYGONS, id="polygons",),
-        pytest.param(H.TABLE, id="table"),
-    ],
-)
-def test_unsupported_merge_conflicts(
-    data, data_working_copy, geopackage, cli_runner, insert
-):
-    with data_working_copy(data.ARCHIVE) as (repo_path, wc):
-        repo = pygit2.Repository(str(repo_path))
-
-        # new branch
-        r = cli_runner.invoke(["checkout", "-b", "alternate"])
-        assert r.exit_code == 0, r
-        assert repo.head.name == "refs/heads/alternate"
-
-        db = geopackage(wc)
-        alternate_commit_id = insert(db, reset_index=1, insert_str="aaa")
-
-        assert repo.head.target.hex == alternate_commit_id
-
-        r = cli_runner.invoke(["checkout", "master"])
-        assert r.exit_code == 0, r
-        assert repo.head.target.hex != alternate_commit_id
-
-        master_commit_id = insert(db, reset_index=1, insert_str="mmm")
-
-        r = cli_runner.invoke(["merge", "alternate"])
-        assert r.exit_code == NOT_YET_IMPLEMENTED
-        assert (
-            "resolving conflicts where features are added or removed isn't supported yet"
-            in r.stderr
-        )
+        if output_format == "text":
+            assert r.stdout.split("\n") == [
+                'Merging branch "alternate" into master',
+                "Conflicts found:",
+                "",
+                f"{data.LAYER}:",
+                "  Feature conflicts:",
+                "    edit/edit: 3",
```

Comment:
Feels like this should list what the conflicts actually are? rather than just counting them by type

Reply:
Maybe - inclined to do this in a follow up PR though.
Certainly we need to have the ability to display all of the conflicts, otherwise the user can't resolve them. And, we likely need a separate command to show conflicts, because the user if the user returns to half-finished merge, they will want to see them again. So, if we do want to display all resulting conflicts in full in the `sno merge` output, it makes sense to do it by delegating to that other (as yet unimplemented ) command.

Counting them by type is still useful because they will probably still need to be sorted by type. Eventually (when we support this) the user will be required to resolve schema conflicts for a particular table before resolving feature conflicts. Until the resolved schema is known, it is not clear how the features look, or if they are even still conflicts which need resolving.

Reply:
> in a follow up PR
ok

> we likely need a separate command to show conflicts, because the user if the user returns to half-finished merge, they will want to see them again

IMHO, `sno status` should do that, if you're halfway through a merge